### PR TITLE
[Documentation] Squiz.WhiteSpace.FunctionClosingBraceSpace

### PR DESCRIPTION
**Pull request by przemekhernik**
## Description

This PR adds documentation for the `Squiz.WhiteSpace.FunctionClosingBraceSpace` [sniff](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/src/Standards/Squiz/Sniffs/WhiteSpace/FunctionClosingBraceSpaceSniff.php).

- `bin/phpcs --standard=Squiz --generator=text --sniffs=Squiz.WhiteSpace.FunctionClosingBraceSpace`
- `bin/phpcs --standard=Squiz --generator=html --sniffs=Squiz.WhiteSpace.FunctionClosingBraceSpace`

image
image

## Suggested changelog entry

Add documentation for the `Squiz.WhiteSpace.FunctionClosingBraceSpace` sniff.

## Related issues/external references

Part of https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/148

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [x] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <![CDATA[
+    There should be a single blank line before the closing function brace.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: A single blank line before the closing function brace.">
+        <![CDATA[
+function outer() <em>{
+
+}</em>;
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:12]:
```suggestion
}</em>
```

Stray semicolon. This is a named function, not a closure, so no need for the semicolon.

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+]]>
+        </code>
+        <code title="Invalid: No blank line before the closing function brace.">
+        <![CDATA[
+function outer() <em>{}</em>;
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:17]:
```suggestion
function outer() <em>{}</em>
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: Nested function closing brace in the new line.">
+        <![CDATA[
+function outer() {
+    function inner() <em>{
+    }</em>;
+
+};
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:34]:
```suggestion
function outer() {
    function inner() <em>{
    }</em>

}
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+]]>
+        </code>
+        <code title="Invalid: Nested function closing brace in the same line.">
+        <![CDATA[
+function outer() {
+    function inner() <em>{}</em>;
+
+};
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:42]:
```suggestion
function outer() {
    function inner() <em>{}</em>

}
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+<documentation title="Function Closing Brace Space">
+    <standard>
+    <![CDATA[
+    There should be a single blank line before the closing function brace.
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:4]:
```suggestion
    There should be a single blank line before the function closing brace.
```

Reason: consistency with the sniff name and readability.

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: A single blank line before the closing function brace.">
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:8]:
```suggestion
        <code title="Valid: A single blank line before the function closing brace.">
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        <![CDATA[
+function outer() <em>{
+
+}</em>;
+]]>
+        </code>
+        <code title="Invalid: No blank line before the closing function brace.">
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:15]:
```suggestion
        <code title="Invalid: No blank line before the function closing brace.">
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        </code>
+    </code_comparison>
+
+    <standard>
+    <![CDATA[
+    Closing brace of nested function must be on a new line.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: Nested function closing brace in the new line.">
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:28]:
```suggestion
        <code title="Valid: Nested function closing brace is on a new line.">
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        <![CDATA[
+function outer() {
+    function inner() <em>{
+    }</em>;
+
+};
+]]>
+        </code>
+        <code title="Invalid: Nested function closing brace in the same line.">
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:37]:
```suggestion
        <code title="Invalid: Nested function closing brace on the same line.">
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    function inner() <em>{}</em>;
+
+};
+]]>
+        </code>
+    </code_comparison>
+
+    <standard>
+    <![CDATA[
+    Closing brace of nested function must not be followed by blank line.
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:49]:
```suggestion
    Closing brace of nested function must not be followed by a blank line.
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: No blank line before nested function closing brace.">
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:53]:
```suggestion
        <code title="Valid: No blank line before the closing brace of a nested function.">
```

File: src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+        <![CDATA[
+function outer() {
+    function inner() <em>{
+    }</em>
+
+}
+]]>
+        </code>
+        <code title="Invalid: A single blank line before nested function closing brace.">
```
Comment (jrfnl) [inline src/Standards/Squiz/Docs/WhiteSpace/FunctionClosingBraceSpaceStandard.xml:62]:
```suggestion
        <code title="Invalid: A single blank line before the closing brace of a nested function.">
```

**Review by jrfnl — COMMENTED** [review]
@przemekhernik Thank you for this PR and my apologies that it took me a while before I got to reviewing it.

Overall, this looks good. All errors accounted for and the code samples demonstrate the issues clearly.

There are just some grammatical nitpicks to fix and then I'm happy to merge this.

**Comment by jrfnl** [comment]
@przemekhernik Just checking in to see when you have time to finish this off ... ?

**Comment by przemekhernik** [comment]
Hi @jrfnl ! I'm sorry for the delay too! I completely missed the email with the notification. I've applied your great suggestions and it's ready to review 🤘

**Review by jrfnl — APPROVED** [review]
Thanks @przemekhernik for those updates and for contributing.

All good now. I will rebase the PR (without changes) and merge once the build has passed.

**Pull request merged**